test_conformance/api/test_sub_group_dispatch.cpp

```diff
@@ -75,6 +75,14 @@ int test_sub_group_dispatch(cl_device_id deviceID, cl_context context, cl_comman
     size_t ret_ndrange2d_flattened;
     size_t ret_ndrange3d_flattened;
 
+    // This test should be skipped in the case that the cl_khr_subgroups
+    // extension is not supported.
+    if (!is_extension_available(deviceID, "cl_khr_subgroups"))
```

Comment:
Could you add the logic for checking device version from #864 please?

I created that PR without realising that this one already existed.

Reply:
Thanks for your feedback @Jeremy-Kemp. I think this PR has been superceded by #816 which handles optional features in a more generic and fashion. I was waiting for #816 to be merged before closing this, but I'm also happy to close this in favour of #816 since you have already implemented the check for the device version and the core feature.

Reply:
@FranklandJack It will be eventually but I've been struggling to find the time to spend on #816 and this is starting to get in the way of progress so I suggested we merge your PRs / Jeremy's and I'll rework this when I work on #816 again.

Reply:
@kpet Okay that sounds like a good plan to me. I've updated #789 this morning and am happy to close this PR in favor of @Jeremy-Kemp 's #864 since the work is already done there.